pkgs/development/tools/watchman/default.nix

```diff
@@ -1,46 +1,86 @@
-{ stdenv, lib, config, fetchFromGitHub, autoconf, automake, pcre
-, libtool, pkg-config, openssl
+{ stdenv, lib, config, fetchFromGitHub, pcre, cmake, folly, fizz, libsodium, fb303, cpptoml, rustPlatform, libunwind, double-conversion, lz4, zstd
+, pkg-config, openssl, python3, gtest, glog, boost, libevent, fmt_9, zlib, wangle, fbthrift, edencommon, ensureNewerSourcesHook
 , confFile ? config.watchman.confFile or null
 , withApple ? stdenv.isDarwin, CoreServices
 }:
 
 stdenv.mkDerivation rec {
   pname = "watchman";
-  version = "4.9.0";
+  version = "2022.07.25.00";
 
   src = fetchFromGitHub {
     owner = "facebook";
     repo = "watchman";
     rev = "v${version}";
-    sha256 = "0fdaj5pmicm6j17d5q7px800m5rmam1a400x3hv1iiifnmhgnkal";
+    sha256 = "sha256-ewneXWWdIPBNjkJzEfxCSLQmG6fJ3//1C1BZTIRnk8A=";
   };
 
-  nativeBuildInputs = [ autoconf automake pkg-config libtool ];
-  buildInputs = [ pcre openssl ]
-    ++ lib.optionals withApple [ CoreServices ];
+  cmakeFlags = [
+    "-DBUILD_SHARED_LIBS=ON"
+    "-DENABLE_EDEN_SUPPORT=NO"
+    "-DWATCHMAN_STATE_DIR=/tmp/watchman-state"
+  ];
+
+  nativeBuildInputs = [
+    cmake
+    pkg-config
+    (ensureNewerSourcesHook {
+      year = "1980";
+    })
+  ] ++ (with rustPlatform; [
+    cargoSetupHook
+    rust.cargo
+    rust.rustc
+  ]);
+
+  buildInputs = [
+    pcre
+    openssl
+    python3
+    gtest
+    glog
+    boost
+    libevent
+    fmt_9
+    libsodium
+    zlib
+    folly
+    fizz
+    wangle
+    fbthrift
+    fb303
+    cpptoml
+    edencommon
+    libunwind
+    double-conversion
+    lz4
+    zstd
+  ] ++ lib.optionals withApple [ CoreServices ];
 
   configureFlags = [
     "--enable-lenient"
     "--enable-conffile=${if confFile == null then "no" else confFile}"
     "--with-pcre=yes"
-
-    # For security considerations re: --disable-statedir, see:
-    # https://github.com/facebook/watchman/issues/178
-    "--disable-statedir"
   ];
 
   prePatch = ''
     patchShebangs .
   '';
 
-  preConfigure = ''
-    ./autogen.sh
-  '';
+  patches = [
+    ./cargo-lock.patch
+  ];
```

Comment:
Why do we need the lock file twice? Can we fetchCrate instead?

Reply:
Fixed